runtime/src/snapshot_info.rs

```diff
@@ -0,0 +1,173 @@
+//! Shared runtime information about snapshots
+//!
+//! Snapshot information is needed at runtime in multiple threads.  Currently, this is used to
+//! prevent cleaning of zero-lamport accounts in `AccountsDb` _past_ the last full snapshot slot.
+//! However, snapshots are taken in a separate thread, and previously `AccountsDb` didn't have a
+//! way to get that information.
+//!
+//! This module was created to solve that problem.  Now, all threads that need runtime snapshot
+//! information can have a `SyncSnapshotInfo` object to get/set the information they need.
+//!
+//! **NOTE** Client should use `SyncSnapshotInfo` and not `SnapshotInfo`, since the later does not
+//! include synchronization (and the whole point of this module is to share snapshot information
+//! between threads).
+//!
+//! **NOTE** There are helper functions to handle the common case of getting/setting snapshot
+//! information when you have a `Option<SyncSnapshotInfo>`.
+
+use solana_sdk::clock::Slot;
+use std::sync::{Arc, RwLock};
+
+/// Runtime information about snapshots
+///
+/// **NOTE** Clients should use `SyncSnapshotInfo` instead!
+///
+/// Information about the snapshots is needed in multiple threads at different times; for example,
+/// `AccountsDb` (and `Bank` via `AccountsDb`) need to ensure cleaning does not happen past the
+/// `last_full_snapshot_slot`.  Whereas `snapshot_utils`, `serde_snapshot`, `AccountsBackgroundService`, and
+/// `SnapshotPackagerService` need to be able to set these values when snapshots are taken.
+///
+/// Share this object between all interested parties to ensure a consistent view of the snapshot
+/// information.
+#[derive(Debug)]
+pub struct SnapshotInfo {
```

Comment:
SnapshotInfo can be easily to read as info about a particular snapshot. Maybe SnapshotRuntimeInfo or LastSnapshotInfo is more appropriate?

Reply:
After I put up this PR, then added `SnapshotArchiveInfo` in a different PR, I had the same thought for this one. I'll rename it `SnapshotRuntimeInfo`.

Reply:
Done!